core/src/main/java/org/elasticsearch/plugins/DiscoveryPlugin.java

```diff
@@ -52,4 +56,6 @@
     default NetworkService.CustomNameResolver getCustomNameResolver(Settings settings) {
         return null;
     }
+
+    default Set<Class<? extends ZenPing>> getZenPings(Settings settings) { return Collections.emptySet(); }
```

Comment:
Look at other methods that are overriden in MockNode. Essentially, create a method on Node to create the ZenPing. Override it in mock node to return something different than the super method does.